### PR TITLE
integrate T2-Adapter into SDXL

### PR DESCRIPTION
**Pull request by hadipash**
changelist:
- Extracted T2I-Adapter into a separate directory as it works with SD and SDXL.
- Added T2I-Adapters support for SDXL.

### CODE REVIEW

**Review by SamitHuang — APPROVED** [review]

**Comment by SamitHuang** [comment]
Please resolve the conflicts

File: examples/stable_diffusion_xl/gm/modules/diffusionmodules/openaimodel.py [hunk 1]
```diff
@@ -787,11 +789,26 @@ def construct(self, x, timesteps=None, context=None, y=None, **kwargs):
...
+                h += adapter_states[adapter_idx]
+                adapter_idx += 1
+
             hs_idx += 1
+
         h = self.middle_block(h, emb, context)
+        if adapter_states:
+            h += adapter_states[adapter_idx]
+
         for module in self.output_blocks:
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/modules/diffusionmodules/openaimodel.py:812]: Perhaps we should add a new class that inherits from UNetModel to manage the adapter. The addition of new branches may affect the original compilation process and time.

File: examples/stable_diffusion_xl/gm/helpers.py [hunk 2]
```diff
@@ -401,6 +412,7 @@ def init_sampling(
     num_cols=None,
     sampler="EulerEDMSampler",
     guider="VanillaCFG",
+    guidance_scale=5.0,
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/helpers.py:415]: Please also modify in `./demo/streamlit_helpers.py`

File: examples/stable_diffusion_xl/README.md [hunk 3]
```diff
@@ -15,6 +15,9 @@ This folder contains [Stable Diffusion XL (SDXL)](https://arxiv.org/abs/2307.019
 
 ## What is New
 
+**Nov 20, 2023**
+1. Add [T2I-Adapters](../t2i_adapter/README.md) support for SDXL.
+
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/README.md:20]: Please also add an explanation in features and getting_strated.md.

**Comment by zhanghuiyao** [comment]
Please resolve the file conflict

File: examples/stable_diffusion_xl/gm/helpers.py (hunk 2 above)
Comment (hadipash) [reply examples/stable_diffusion_xl/gm/helpers.py:415]: Not sure what you mean, `stable_diffusion_xl/demo/streamlit_helpers.py` [already has](https://github.com/mindspore-lab/mindone/blob/7e4f9cf9bdfeeac4e0877730d9a62b934adba688/examples/stable_diffusion_xl/demo/streamlit_helpers.py#L144) the guidance scale.

File: examples/stable_diffusion_xl/gm/modules/diffusionmodules/openaimodel.py (hunk 1 above)
Comment (hadipash) [reply examples/stable_diffusion_xl/gm/modules/diffusionmodules/openaimodel.py]: Good suggestion. Extracted T2I-Adapter's UNet in a separate module.

**Review by zhtmike — APPROVED** [review]

**Pull request merged**